File: APP MOBILE/smart_money/lib/modules/auth/models/reset_password_request.dart
/// Request đặt lại mật khẩu (sau khi có OTP).
/// Tương ứng: ResetPasswordRequest.java (server)
class ResetPasswordRequest {
  final String email;
  final String otp;
  final String newPassword;

  const ResetPasswordRequest({
    required this.email,
    required this.otp,
    required this.newPassword,
  });

  Map<String, dynamic> toJson() => {
        'email': email,
        'otp': otp,
        'newPassword': newPassword,
      };
}

File: smartmoneyserver/src/main/java/fpt/aptech/server/dto/request/ResetPasswordRequest.java
package fpt.aptech.server.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ResetPasswordRequest {
    @NotBlank(message = "Email cannot be empty")
    private String email;

    @NotBlank(message = "OTP code cannot be empty")
    private String otp;

    @NotBlank(message = "New password cannot be empty")
    @Size(min = 6, message = "Password must be at least 6 characters")
    private String newPassword;
}